### ISSUE TITLE
untokenize of specially crafted escaped characters does not round trip properly

### ISSUE THREAD

**Issue by asottile** [python/cpython#125821]
# Bug report

### Bug description:

this small program does not roundtrip through `tokenize` / `untokenize` -- it appears to mishandle the escaped quote as a `\N{NAMED ESCAPE}`

```python
bar = 1
print(f"{bar} \"{{SNOWMAN}} {{foo}}")
```

this is what it produces after a round of untokenization:

```console
$ python3 t.py t3.py 
bar = 1
print(f"{bar} \"{ SNOWMAN}} {{foo}}")
```

annoyingly, `tokenize_rt` suffers from a different related bug which is why I was investigating this to begin with.  an aside, the handling of curly braces in 3.12+ tokenization is a huge pain!

### CPython versions tested on:

CPython main branch

### Operating systems tested on:

Linux

**Comment by asottile** [comment]
cc @pablogsal ecf16ee50e42f979624e55fa343a8522942db2e7

**Comment by pablogsal** [comment]
Sorry is a bit late here and I had a long day, but what I am missing here:

```
>>> import tokenize
>>> import io
>>> code = r'f"{bar} \"{{SNOWMAN}} {{foo}}"'
>>> tokens = list(tokenize.generate_tokens(io.StringIO(code).readline))
>>> tokenize.untokenize(tokens)
'f"{bar} \\"{{SNOWMAN}} {{foo}}"'
>>> code
'f"{bar} \\"{{SNOWMAN}} {{foo}}"'
zzz

**Comment by pablogsal** [comment]
Duplicate of https://github.com/python/cpython/issues/125008

**Comment by asottile** [comment]
hah yep -- just checked my vod and I spent all the time waiting on a build only to rerun my system python 🤦
